File: app/src/main/java/com/hencoder/hencoderpracticedraw2/practice/Practice03SweepGradientView.java
package com.hencoder.hencoderpracticedraw2.practice;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Shader;
import android.graphics.SweepGradient;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;

public class Practice03SweepGradientView extends View {

    private Paint mPaint;

    public Practice03SweepGradientView(Context context) {
        this(context, null);
    }

    public Practice03SweepGradientView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public Practice03SweepGradientView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public void init() {
        mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        Shader shader = new SweepGradient(300, 300, 0XFFE91E63, 0XFF2196F3);
        mPaint.setShader(shader);
        // 用 Paint.setShader(shader) 设置一个 SweepGradient
        // SweepGradient 的参数：圆心坐标：(300, 300)；颜色：#E91E63 到 #2196F3
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);

        canvas.drawCircle(300, 300, 200, mPaint);
    }
}

File: app/src/main/java/com/hencoder/hencoderpracticedraw2/practice/Practice03SweepGradientView.kt
package com.hencoder.hencoderpracticedraw2.practice

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.SweepGradient
import android.util.AttributeSet
import android.view.View

/**
 * Created by xh on 2017/10/17.
 */
class Practice03SweepGradientView : View {

    private val mPaint = Paint(Paint.ANTI_ALIAS_FLAG)

    private var mSweerpGradient: SweepGradient

    private val mRadius = 200f

    private val mStartColor = 0XFFE91E63.toInt()
    private val mEndColor = 0XFF2196F3.toInt()

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    init {
        mSweerpGradient = SweepGradient(300f, 300f, mStartColor, mEndColor)
        mPaint.shader = mSweerpGradient
    }

    override fun onDraw(canvas: Canvas?) {
        super.onDraw(canvas)

        canvas!!.drawCircle(300f, 300f, mRadius, mPaint)

    }

}